src/app/file/bmp_format.cpp

```diff
@@ -151,26 +242,79 @@ static int read_bmfileheader(FILE *f, BITMAPFILEHEADER *fileheader)
 /* read_win_bminfoheader:
  *  Reads information from a BMP file header.
  */
-static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader)
+static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader, int biSize)
 {
   WINBMPINFOHEADER win_infoheader;
 
+  switch (biSize) {
+    case 16:
+    case 24:
+    case WININFOHEADERSIZE: // 40
+    case BV2INFOHEADERSIZE: // 52
+    case BV3INFOHEADERSIZE: // 56
+    case 60:
+    case 64:
+      break;
+    default:
+      return -1;
+  }
```

Comment:
What do you think about refactoring the switch case as follows?
```suggestion
  if (biSize != 16 && 
      biSize != 24 && 
      biSize != WININFOHEADERSIZE && 
      biSize != BV2INFOHEADERSIZE &&
      biSize != BV3INFOHEADERSIZE && 
      biSize != 60 && 
      biSize != 64) return -1;
```

Reply:
Done